Fitness/Controllers/HomeViewController+Extensions.swift

```diff
@@ -0,0 +1,224 @@
+        favoriteGyms = favorites.compactMap { favorite in
+            self.gyms.first { $0.name == favorite }
+        }
+    }
+
+    func pushHabitOnboarding() {
+        let habitViewController = HabitTrackingController(type: .cardio)
+        self.navigationController?.pushViewController(habitViewController, animated: true)
+        return
+    }
+
+    func pushGymOnboarding() {
+        navigationController?.tabBarController?.tabBar.isHidden = true
+        let onboardingGymsViewController = OnboardingGymsViewController()
+        onboardingGymsViewController.delegate = self
+        navigationController?.pushViewController(onboardingGymsViewController, animated: true)
+        return
+    }
+
+    func viewTodaysClasses() {
+        guard let classNavigationController = tabBarController?.viewControllers?[1] as? UINavigationController else { return }
+        guard let classListViewController = classNavigationController.viewControllers[0] as? ClassListViewController else { return }
+
+        classListViewController.updateCalendarDateSelectedToToday()
+
+        classNavigationController.setViewControllers([classListViewController], animated: false)
+        tabBarController?.hidesBottomBarWhenPushed = false
+        tabBarController?.selectedIndex = 1
+    }
+
+    func isGymStatusChangingSoon(gym: Gym) -> Bool {
+        let changingSoonThreshold = 3600.0
+        let now = Date()
+
+        if gym.isOpen {
+            return (gym.gymHoursToday.closeTime - changingSoonThreshold) < now
+        } else {
+            let openTime = gym.gymHours[now.getIntegerDayOfWeekTomorrow()].openTime + Date.secondsPerDay
+            return (openTime - changingSoonThreshold) < now
+        }
+    }
+
+    func getHourString(gym: Gym) -> String {
+        let now = Date()
+        let isOpen = gym.isOpen
+
+        let gymHoursToday = gym.gymHoursToday
+        let gymHoursTomorrow = gym.gymHours[now.getIntegerDayOfWeekTomorrow()]
+
+        let format: String
+
+        if now > gymHoursToday.closeTime {
+            if Calendar.current.component(.minute, from: gymHoursTomorrow.openTime) == 0 {
+                format = "h a"
+            } else {
+                format = "h:mm a"
+            }
+            if gym.closedTomorrow {
+                return "Closed Tomorrow"
+            } else {
+                return "Opens at \(gymHoursTomorrow.openTime.getStringOfDatetime(format: format))"
+            }
+        } else if !isOpen {
+            if Calendar.current.component(.minute, from: gymHoursToday.openTime) == 0 {
+                format = "h a"
+            } else {
+                format = "h:mm a"
+            }
+            return "Opens at \(gymHoursToday.openTime.getStringOfDatetime(format: format))"
+        } else {
+            if Calendar.current.component(.minute, from: gymHoursToday.closeTime) == 0 {
+                format = "h a"
+            } else {
+                format = "h:mm a"
+            }
```

Comment:
maybe factor this out into a really small helper function that takes the `from` as the function argument?

Reply:
good idea will do
i didn't write this :(